Update gRPC probes docs for GA (#39919)
kubernetes/website: content/en/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md

{{< feature-state for_k8s_version="v1.24" state="beta" >}}

If your application implements the [gRPC Health Checking Protocol](https://github.com/grpc/grpc/blob/master/doc/health-checking.md),

> you can configure Kubernetes to use it for application liveness checks.

Reviewer: GRPC probes are only for liveness checks, right?

Reviewer: Ah ... the question is answered later. So ... we may want to revise this statement for accuracy.

Author: rephrased

Here is an example manifest:

{{< codenew file="pods/probe/grpc-liveness.yaml" >}}

To use a gRPC probe, `port` must be configured. You can also use the `service` field

> to distinguish probes of different types and probes for different features.

Reviewer: Sounds like `service` is not a mandatory field.
What happens if it is omitted then?

Author: rephrased

> If your application implements [gRPC Health Checking Protocol](https://github.com/grpc/grpc/blob/master/doc/health-checking.md),

Reviewer 2: Good to fix this for GA:
```suggestion
If your application implements the [gRPC Health Checking Protocol](https://github.com/grpc/grpc/blob/master/doc/health-checking.md),
```

> kubelet can be configured to use it for application liveness checks.

Reviewer 2: ```suggestion
you can configure Kubernetes to use it for application liveness checks.
```

Here is an example manifest:

{{< codenew file="pods/probe/grpc-liveness.yaml" >}}

> To use a gRPC probe, `port` must be configured. You can also use `service` argument

Reviewer 2: ```suggestion
To use a gRPC probe, `port` must be configured. You can also use the `service` field
```

to distinguish probes of different types and probes for different features.
You can set `service` to the value `liveness` and make your gRPC Health Checking endpoint

> respond to this request differently then when `service` set to `readiness`.

Reviewer 2: ```suggestion
respond to this request differently then when you set `service` set to `readiness`.
This lets you use the same endpoint for different kinds of container health check
(rather than needing to listen on two different ports).
```

> If you want to specify service and a probe type, the recommendation is to combine

Reviewer 2: ```suggestion
If you want to specify your own custom service name and also specify a probe type,
the Kubernetes project recommends that you use a name that concatenates
```

> those like `myservice-liveness`.

Reviewer 2: ```suggestion
those. For example: `myservice-liveness` (using `-` as a separator).
```

{{< note >}}

> Unlike HTTP and TCP probes, named ports cannot be used and custom host cannot be configured.

Reviewer 2: ```suggestion
Unlike HTTP or TCP probes, you cannot specify the healthcheck port by name, and you
cannot configure a custom hostname.
```
